docs/topics/multiplatform-mobile/multiplatform-mobile-upgrade-app.md

```diff
@@ -286,89 +486,258 @@ is already imported and used in `ContentView.swift` with `import shared`.
+
+    This will install the KMP-NativeCoroutines packages necessary to work with the `async/await` mechanism.
+
+##### Consume the flow using the KMP-NativeCoroutines library
+
+1. In `iosApp/ContentView.swift`, update the `startObserving()` function to consume the flow using KMP-NativeCoroutine's
+   `asyncSequence()` function for the `Greeting().greet()` function.
+   
+   Use a loop and the `await` mechanism to iterate through the flow and update the `greetings` property every time
+   the flow emits a value.
+2. To support concurrency, wrap the asynchronous operation in a `Task`:
+
+    ```Swift
+    func startObserving() {
+        Task {
+           do {
+               let sequence = asyncSequence(for: Greeting().greet())
+               for try await phrase in sequence {
+                  self.greetings.append(phrase)
+               }
+           } catch {
+               print("Failed with error: \(error)")
+           }
        }
-   }
-   ```
+    }
+    ```
+
+3. Make sure `ViewModel` is marked with the `@MainActor` annotation. It ensures all asynchronous operations within
+   `ViewModel` run on the main thread to comply with the Kotlin/Native requirement:
+
+    ```Swift
+    // ...
+    import KMPNativeCoroutinesAsync
+    import KMPNativeCoroutinesCore
+    
+    // ...
+    extension ContentView {
+        @MainActor
+        class ViewModel: ObservableObject {
+            @Published var greetings: Array<String> = []
+            
+            func startObserving() {
+                Task {
+                   do {
+                       let sequence = asyncSequence(for: Greeting().greet())
+                       for try await phrase in sequence {
+                           self.greetings.append(phrase)
+                       }
+                   } catch {
+                       print("Failed with error: \(error)")
+                   }
+               }
+            }
+        }
+    }
+    ```
 
-6. Re-run both **androidApp** and **iosApp** configurations from Android Studio to make sure your app's logic is synced:
+4. Re-run both **androidApp** and **iosApp** configurations from Android Studio to make sure your app's logic is synced:
 
-   ![Final results](multiplatform-mobile-upgrade.png){width="500"}
+    ![Final results](multiplatform-mobile-upgrade.png){width=500}
 
-> You can find this state of the project in our [GitHub repository](https://github.com/kotlin-hands-on/get-started-with-kmp).
-> 
+<!-- You can find this state of the project in our [GitHub repository](https://github.com/kotlin-hands-on/get-started-with-kmp). -->
+
+#### Option 2. Configure SKIE {initial-collapse-state="collapsed"}
+
+> We recommend using the latest version of the library.
+> Check the [SKIE repository](https://github.com/touchlab/SKIE/releases) to see if a newer version of the plugin is available.
+>
 {type="note"}
 
+To set up the library, in `shared/build.gradle.kts`, specify the SKIE plugin and click the **Sync** button.
```

Comment:
Will it be better to put `shared/build.gradle.kts` in the end of the sentence?

Reply:
it will, thank you!